onset_maproom/agronomy.py

```diff
@@ -1,10 +1,12 @@
 import xarray as xr
+import numpy as np
+
 
 def soil_plant_water_bucket(
```

Comment:
Find a better name? Maybe something with the word "step" in it?

"Bucket" is kind of cute, because buckets hold water, but kind of confusing for the same reason.

Reply:
Will the user of the library need to know call this function, or is it an implementation detail? If an implementation detail, let's make it "private" by either starting its name with an underscore, or adding an `__all__` list and not including this in it.

Reply:
bucket is cute, but is also how they call it, e.g.
https://pressbooks.bccampus.ca/geoglabs2020/wp-content/uploads/sites/1032/2020/08/Figure-8.1-350x292.png
more at:
https://duckduckgo.com/?q=soil+plant+water+balance+bucket+algorithm&t=ffsb&iax=images&ia=images

Reply:
Can we answer the public/private question at the end? I assume private but I don't have a good vision of those things just yet. But yeah I am guessing that at best, we might want to program another kind of step if there are other models than this one, and then we would have some options in the loop function to choose which step-model to use. Or possibly also another loop-function if not compatible.

Reply:
Oh, then I don't understand why you used this word here. Bucket seems to refer here to a volume of space, whereas this function implements a step in time.

Reply:
> Can we answer the public/private question at the end? 

Sure.

Reply:
you do one bucket at a time...! + it's cute indeed. Anyway, it's called _step now